### ISSUE TITLE
Metric version value passed into a `snapctl metric get` command is not checked 

### ISSUE THREAD

**Issue by tjmcs** [intelsdi-x/snap#1082]
With the recent merge of #1076 and the pending merge of #1049, there is potential confusion surrouding the value passed in for the `--metric-version` flag. In short, the values used to retrieve **all** versions of a given metric and the values used to return the **latest** version of a given metric are different depending on whether the `--metric-namespace` value passed into the `snapctl metric get ...` command contains a wildcard or not.

To hide this inconsistency (which is required for historical reasons that we won't elaborate on here) we should add value checking to the value passed in for the `--metric-version` flag (ensuring that it is an integer value and that it is greater than zero). This will prevent users from specifying values that are zero or less than zero for this flag and prevent confusion amongst the users as to why the values used to represent a request to return **all** versions is different depending on whether the `--metric-namespace` value includes a wildcard or not.

While we are making this change, we should also modify the help text associated with the `snapctl metric get ...` command to note that the value passed in for the `--metric-version` must be an integer that is greater than zero (and remove references to the default value?).

**Comment by candysmurf** [comment]
@tjmcs, do you mind to give a couple of examples to demonstrate the issue here? thank you.

**Comment by tjmcs** [comment]
I would have thought it was fairly obvious from my description above, but here's my best shot at a couple of examples. Once #1049 is merged, the specifying a `--metric-version` value that is less than or equal to zero in a `snapctl metric get ...` command that retrieves a single metric will return the latest version of that metric:
```bash
$ snapctl -u "https://127.0.0.1:12345" --insecure metric get --metric-namespace /intel/mock/foo --metric-version 0
NAMESPACE 		 VERSION 	 UNIT 		 LAST ADVERTISED TIME 		 DESCRIPTION
/intel/mock/foo 	 2 		 mock unit 	 Fri, 15 Jul 2016 21:20:47 PDT 	 mock description

  Rules for collecting /intel/mock/foo:

       NAME 		 TYPE 		 DEFAULT 	 REQUIRED 	 MINIMUM 	 MAXIMUM
       name 		 string 	 bob 		 false
       password 	 string 	  		 true

$ snapctl -u "https://127.0.0.1:12345" --insecure metric get --metric-namespace /intel/mock/foo --metric-version '-1'
NAMESPACE 		 VERSION 	 UNIT 		 LAST ADVERTISED TIME 		 DESCRIPTION
/intel/mock/foo 	 2 		 mock unit 	 Fri, 15 Jul 2016 21:20:47 PDT 	 mock description

  Rules for collecting /intel/mock/foo:

       NAME 		 TYPE 		 DEFAULT 	 REQUIRED 	 MINIMUM 	 MAXIMUM
       name 		 string 	 bob 		 false
       password 	 string 	  		 true

$
```
but based on the changes we just merged in PR #1076 a very different result will be returned if a `--metric-version` value of zero is used than will be returned if a `--metric-value` version that is less than zero is used with a command that includes a wildcard in the `--metric-namespace` value once the changes in PR #1049 are merged. At that time, this command:
```bash
$ snapctl -u "https://127.0.0.1:12345" --insecure metric get --metric-namespace '/intel/mock/*' --metric-version 0
```
will return a list of **all** versions of **all** of the metrics under the `/intel/mock/*` namespace, but this command:
```bash
$ snapctl -u "https://127.0.0.1:12345" --insecure metric get --metric-namespace '/intel/mock/*' --metric-version '-1'
```
will only return the **latest** versions of those same metrics. The only way I can see to keep the output consistent in these `snapctl metric get ...` commands (between `--metric-namespace` values that include a wildcard and those that don't) is to prevent users from entering a value that is less than or equal to zero as the `--metric-version` value for this `snapctl` command.

Does that clarify the issue (and what I think we need to do to resolve it), @candysmurf?

**Comment by candysmurf** [comment]
@tjmcs, thanks for your detailed explanation! It's clear now. Thanks for your suggestion to make the behavior consistent!

**Comment by tjmcs** [comment]
With the merge of PR #1049, we can now mark this issue as resolved...after looking through the comments on this story I think it'll be better to open a separate issue up that just addresses the question of documenting the `--metric-version` values that can be used with a `snapctl metric get` command (and the associated `GET` against the `/metrics` RESTful API resource)
